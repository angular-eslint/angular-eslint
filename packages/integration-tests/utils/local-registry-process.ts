/* eslint-disable @typescript-eslint/no-non-null-assertion */
import type { ChildProcess } from 'child_process';
import { spawn } from 'child_process';
import execa from 'execa';
import { join } from 'path';
import kill from 'tree-kill';

export const FIXTURES_DIR = join(__dirname, '../fixtures/');

// Used to ensure all the time-consuming setup steps for fixtures do not cause jest to time out
export const LONG_TIMEOUT_MS = 600000; // 10 mins

let localRegistryProcess: ChildProcess;

const VERSION = `9999.0.1-local-integration-tests`;

const cwd = process.cwd();

process.env.npm_config_registry = `http://localhost:4872/`;
process.env.YARN_REGISTRY = process.env.npm_config_registry;

export async function spawnLocalRegistry(): Promise<void> {
  localRegistryProcess = spawn('npx', [
    'verdaccio',
    '--config',
    './local-registry/config.yml',
    '--listen',
    '4872',
  ]);

  const collectedOutput: string[] = [];
  let resolvedOrRejected = false;

  setTimeout(() => {
    if (!resolvedOrRejected) {
      console.error(`Failed to start the npm registry`);
      console.error(collectedOutput.join(''));
      cleanUp(1);
    }
  }, 10000);

  await new Promise((res, rej) => {
    localRegistryProcess.stdout!.pipe(process.stdout);
    localRegistryProcess.stdout!.on('data', (data: Buffer) => {
      collectedOutput.push(data.toString());
      // wait for local-registry to come online
      if (data.includes('http address')) {
        resolvedOrRejected = true;
        res(undefined);
      }
    });
    localRegistryProcess.stderr!.pipe(process.stderr);
    localRegistryProcess.on('error', (err: Error) => {
      console.error(collectedOutput.join(''));
      resolvedOrRejected = true;
      rej(err);
    });
  });
}

export function cleanUp(code: number): void {
  process.chdir(cwd);

  // try and terminate everything first
  try {
    if (!process.env.CI) {
      kill(0);
    }
    // eslint-disable-next-line no-empty
  } catch {}
  try {
    if (localRegistryProcess) localRegistryProcess.kill();
    // eslint-disable-next-line no-empty
  } catch {}
  // try killing everything after in case something hasn't terminated
  try {
    if (!process.env.CI) {
      kill(0, 'SIGKILL');
    }
    // eslint-disable-next-line no-empty
  } catch {}
  try {
    if (localRegistryProcess) localRegistryProcess.kill('SIGKILL');
    // eslint-disable-next-line no-empty
  } catch {}

  process.exit(code);
}

export async function publishPackagesToVerdaccio(): Promise<
  execa.ExecaChildProcess<string>
> {
  if (process.env.npm_config_registry?.indexOf('http://localhost') === -1) {
    throw Error(`
      ------------------
      ðŸ’£ ERROR ðŸ’£ => $NPM_REGISTRY does not look like a local registry'
      ------------------
    `);
  }

  const subprocess = execa('./publish-to-verdaccio.sh', [VERSION]);
  subprocess.stdout!.pipe(process.stdout);
  subprocess.stderr!.pipe(process.stderr);

  return await subprocess;
}

export async function runNpmInstall(): Promise<
  execa.ExecaChildProcess<string>
> {
  if (process.env.npm_config_registry?.indexOf('http://localhost') === -1) {
    throw Error(`
      ------------------
      ðŸ’£ ERROR ðŸ’£ => $NPM_REGISTRY does not look like a local registry'
      ------------------
    `);
  }

  const subprocess = execa('npm', ['install']);
  subprocess.stdout!.pipe(process.stdout);
  subprocess.stderr!.pipe(process.stderr);

  return await subprocess;
}

export async function runYarnInstall(): Promise<
  execa.ExecaChildProcess<string>
> {
  if (process.env.npm_config_registry?.indexOf('http://localhost') === -1) {
    throw Error(`
      ------------------
      ðŸ’£ ERROR ðŸ’£ => $NPM_REGISTRY does not look like a local registry'
      ------------------
    `);
  }

  const subprocess = execa('yarn', ['install']);
  subprocess.stdout!.pipe(process.stdout);
  subprocess.stderr!.pipe(process.stderr);

  return await subprocess;
}

export async function runNgAdd(): Promise<execa.ExecaChildProcess<string>> {
  if (process.env.npm_config_registry?.indexOf('http://localhost') === -1) {
    throw Error(`
      ------------------
      ðŸ’£ ERROR ðŸ’£ => $NPM_REGISTRY does not look like a local registry'
      ------------------
    `);
  }

  const subprocess = execa('npx', [
    'ng',
    'add',
    `@angular-eslint/schematics@${VERSION}`,
  ]);
  subprocess.stdout!.pipe(process.stdout);
  subprocess.stderr!.pipe(process.stderr);

  return await subprocess;
}

export async function runNgNew(
  workspaceName: string,
  createApplication = true,
): Promise<execa.ExecaChildProcess<string>> {
  if (process.env.npm_config_registry?.indexOf('http://localhost') === -1) {
    throw Error(`
      ------------------
      ðŸ’£ ERROR ðŸ’£ => $NPM_REGISTRY does not look like a local registry'
      ------------------
    `);
  }

  const ngNewArgs = [
    `--strict=true`,
    `--package-manager=npm`,
    `--interactive=false`,
  ];

  if (!createApplication) {
    ngNewArgs.push(`--create-application=false`);
  }

  const subprocess = execa('../../../node_modules/.bin/ng', [
    'new',
    ...ngNewArgs,
    workspaceName,
  ]);
  subprocess.stdout!.pipe(process.stdout);
  subprocess.stderr!.pipe(process.stderr);

  return await subprocess;
}

export async function runNgGenerate(
  args: string[],
): Promise<execa.ExecaChildProcess<string>> {
  if (process.env.npm_config_registry?.indexOf('http://localhost') === -1) {
    throw Error(`
      ------------------
      ðŸ’£ ERROR ðŸ’£ => $NPM_REGISTRY does not look like a local registry'
      ------------------
    `);
  }

  const subprocess = execa('npx', ['ng', 'generate', ...args]);
  subprocess.stdout!.pipe(process.stdout);
  subprocess.stderr!.pipe(process.stderr);

  return await subprocess;
}
