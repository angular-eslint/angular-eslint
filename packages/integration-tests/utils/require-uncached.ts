/**
 * It's important in our snapshot assertions which target modules on disk
 * that we are always grabbing a fresh copy of the file. Node's built-in
 * caching mechanism can get in the way of that so this utility helps
 * us work around the potential issue.
 *
 * (Having the return type as any makes it much easier to work with).
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function requireUncached(modulePath: string): any {
  delete require.cache[require.resolve(modulePath)];
  return require(modulePath);
}
