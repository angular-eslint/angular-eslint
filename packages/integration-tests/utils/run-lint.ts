/* eslint-disable @typescript-eslint/no-non-null-assertion */
import execa from 'execa';
import path from 'path';

import stripAnsi = require('strip-ansi');

const FIXTURES_DIR = path.join(__dirname, '../fixtures/');

function normalizeOutput(value: unknown): string {
  const ansiRemoved = stripAnsi(String(value));
  return (
    ansiRemoved
      .replace(
        new RegExp(`^${FIXTURES_DIR.replace(/\\/g, '\\\\')}(.*?)$`, 'gm'),
        (_, c1) => `__ROOT__/${c1.replace(/\\/g, '/')}`,
      )
      /**
       * This one is ridiculous...
       *
       * No matter what I try (including the use of `strip-ansi`), I do not seem to be
       * able to stop the inclusion of this extra `m` any other way (it is presumably
       * coming from an ANSI code becuase of its initial appearance in CI as [27m).
       * It happens locally and CI... but only exactly once.
       */
      .replace('mmax-len', 'max-len')
  );
}

export async function runLint(directory: string): Promise<string | undefined> {
  try {
    const subprocess = execa('npx', ['ng', 'lint'], {
      cwd: path.join(FIXTURES_DIR, directory),
    });

    /* eslint-disable @typescript-eslint/no-non-null-assertion */
    subprocess.stdout!.pipe(process.stdout);
    subprocess.stderr!.pipe(process.stderr);
    /* eslint-enable @typescript-eslint/no-non-null-assertion */

    const { stdout } = await subprocess;

    return normalizeOutput(stdout);
  } catch (error: any) {
    return normalizeOutput(error.stdout || error);
  }
}
