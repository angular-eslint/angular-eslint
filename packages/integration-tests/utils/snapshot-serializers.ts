function escapeRegExp(str: string) {
  return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
}

function normalizeVersionOfPackage(str: string, pkg: string) {
  const regex = new RegExp(
    `("${escapeRegExp(pkg)}": "[~|^]\\d\\d?\\.)(\\d.\\d)"`,
  );
  return str.replace(regex, '$1X.X"');
}

const dependenciesToNormalize = [
  '@angular-devkit/build-angular',
  '@angular/cli',
  '@angular/compiler-cli',
  'ng-packagr',
  'typescript',
];

/**
 * Normalize dependencies controlled by the Angular CLI so that only the major version is explicitly
 * checked so that we can massively cut down on snapshot update noise across PRs.
 */
export const normalizeVersionsOfPackagesWeDoNotControl = {
  serialize(str: string) {
    for (const pkg of dependenciesToNormalize) {
      str = normalizeVersionOfPackage(str, pkg);
    }
    return str;
  },
  test(val: string) {
    // Only run if we think it's package.json contents
    return (
      val != null &&
      typeof val === 'string' &&
      dependenciesToNormalize.some((pkg) => val.includes(`"${pkg}"`))
    );
  },
};
