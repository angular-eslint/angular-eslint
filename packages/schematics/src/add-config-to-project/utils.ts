/**
 * Utils taken from various parts of Nx:
 * https://github.com/nrwl/nx
 *
 * Thanks, Nrwl folks!
 */
import { normalize } from '@angular-devkit/core';
import { Tree, SchematicContext, Rule } from '@angular-devkit/schematics';
import stripJsonComments from 'strip-json-comments';

/**
 * This method is specifically for reading JSON files in a Tree
 * @param host The host tree
 * @param path The path to the JSON file
 * @returns The JSON data in the file.
 */
function readJsonInTree<T = any>(host: Tree, path: string): T {
  if (!host.exists(path)) {
    throw new Error(`Cannot find ${path}`);
  }
  const contents = stripJsonComments(host.read(path)!.toString('utf-8'));
  try {
    return JSON.parse(contents);
  } catch (e) {
    throw new Error(`Cannot parse ${path}: ${e.message}`);
  }
}

function getWorkspacePath(host: Tree) {
  const possibleFiles = ['/workspace.json', '/angular.json', '/.angular.json'];
  return possibleFiles.filter((path) => host.exists(path))[0];
}

export function getProjectConfig(host: Tree, name: string): any {
  const workspaceJson = readJsonInTree(host, getWorkspacePath(host));
  const projectConfig = workspaceJson.projects[name];
  if (!projectConfig) {
    throw new Error(`Cannot find project '${name}'`);
  } else {
    return projectConfig;
  }
}

export function offsetFromRoot(fullPathToSourceDir: string): string {
  const parts = normalize(fullPathToSourceDir).split('/');
  let offset = '';
  for (let i = 0; i < parts.length; ++i) {
    offset += '../';
  }
  return offset;
}

function serializeJson(json: any): string {
  return `${JSON.stringify(json, null, 2)}\n`;
}

export function updateWorkspaceInTree<T = any, O = T>(
  callback: (json: T, context: SchematicContext, host: Tree) => O,
): Rule {
  return (host: Tree, context: SchematicContext): Tree => {
    const path = getWorkspacePath(host);
    host.overwrite(
      path,
      serializeJson(callback(readJsonInTree(host, path), context, host)),
    );
    return host;
  };
}
