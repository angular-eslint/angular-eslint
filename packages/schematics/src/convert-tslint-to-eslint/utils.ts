import type { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
import * as assert from 'assert';
import { sortObjectByKeys } from '../utils';

export function updateArrPropAndRemoveDuplication(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  json: Record<string, any>,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  configBeingExtended: Record<string, any>,
  arrPropName: string,
  deleteIfUltimatelyEmpty: boolean,
): void {
  json[arrPropName] = json[arrPropName] || [];
  configBeingExtended[arrPropName] = configBeingExtended[arrPropName] || [];
  json[arrPropName] = json[arrPropName].filter(
    (extended: string) => !configBeingExtended[arrPropName].includes(extended),
  );
  json[arrPropName] = Array.from(new Set(json[arrPropName]));
  if (deleteIfUltimatelyEmpty && json[arrPropName].length === 0) {
    delete json[arrPropName];
  }
}

export function updateObjPropAndRemoveDuplication(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  json: Record<string, any>,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  configBeingExtended: Record<string, any>,
  objPropName: string,
  deleteIfUltimatelyEmpty: boolean,
): void {
  json[objPropName] = json[objPropName] || {};
  configBeingExtended[objPropName] = configBeingExtended[objPropName] || {};

  for (const [name, val] of Object.entries(json[objPropName])) {
    const valueOfSamePropInExtendedConfig =
      configBeingExtended[objPropName][name];

    try {
      assert.deepStrictEqual(val, valueOfSamePropInExtendedConfig);
      delete json[objPropName][name];
      // eslint-disable-next-line no-empty
    } catch {}
  }

  if (deleteIfUltimatelyEmpty && Object.keys(json[objPropName]).length === 0) {
    delete json[objPropName];
  }
}

export function ensureESLintPluginsAreInstalled(
  eslintPluginsToBeInstalled: string[],
): Rule {
  return (host: Tree, context: SchematicContext) => {
    if (!eslintPluginsToBeInstalled?.length) {
      return;
    }

    if (!host.exists('package.json')) {
      throw new Error(
        'Could not find a `package.json` file at the root of your workspace',
      );
    }

    const projectPackageJSON = (host.read('package.json') as Buffer).toString(
      'utf-8',
    );
    const json = JSON.parse(projectPackageJSON);
    json.devDependencies = json.devDependencies || {};

    const pluginsToInstall = [];

    for (const pluginName of eslintPluginsToBeInstalled) {
      if (
        !json.devDependencies[pluginName] &&
        !json.dependencies?.[pluginName]
      ) {
        json.devDependencies[pluginName] =
          // Hard code eslint-plugin-jsdoc version for now so that breaking rule change does not apply
          pluginName === 'eslint-plugin-jsdoc' ? '41.1.2' : 'latest';
        pluginsToInstall.push(pluginName);
      }
    }

    if (pluginsToInstall.length > 0) {
      context.logger.info(
        '\nINFO: To most closely match your tslint.json, the `latest` version of the following eslint plugin(s) have been installed:',
      );
      context.logger.info('\n  - ' + pluginsToInstall.join('\n  - '));
      context.logger.info(
        '\nPlease note, you may wish to pin these to a specific version number in your package.json, rather than leaving it open to `latest`.\n',
      );

      host.overwrite('package.json', JSON.stringify(json, null, 2));
      context.addTask(new NodePackageInstallTask());
    }

    return host;
  };
}

export function uninstallTSLintAndCodelyzer(): Rule {
  return (host: Tree, context: SchematicContext) => {
    if (!host.exists('package.json')) {
      throw new Error(
        'Could not find a `package.json` file at the root of your workspace',
      );
    }

    const projectPackageJSON = (host.read('package.json') as Buffer).toString(
      'utf-8',
    );
    const json = JSON.parse(projectPackageJSON);

    if (json.devDependencies) {
      delete json.devDependencies['tslint'];
      delete json.devDependencies['codelyzer'];
      json.devDependencies = sortObjectByKeys(json.devDependencies);
    }

    if (json.dependencies) {
      delete json.dependencies['tslint'];
      delete json.dependencies['codelyzer'];
      json.dependencies = sortObjectByKeys(json.dependencies);
    }

    host.overwrite('package.json', JSON.stringify(json, null, 2));
    context.addTask(new NodePackageInstallTask());

    return host;
  };
}
