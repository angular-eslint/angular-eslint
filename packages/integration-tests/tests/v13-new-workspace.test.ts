/* eslint-disable @typescript-eslint/no-var-requires */
import path from 'path';
import {
  FIXTURES_DIR,
  LONG_TIMEOUT_MS,
  runNgAdd,
  runNgGenerate,
  runNgNew,
} from '../utils/local-registry-process';
import { requireUncached } from '../utils/require-uncached';
import { runLint } from '../utils/run-lint';
import { normalizeVersionsOfPackagesWeDoNotControl } from '../utils/snapshot-serializers';

expect.addSnapshotSerializer(normalizeVersionsOfPackagesWeDoNotControl);

const fixtureDirectory = 'v13-new-workspace';

describe(fixtureDirectory, () => {
  jest.setTimeout(LONG_TIMEOUT_MS);

  beforeEach(async () => {
    process.chdir(FIXTURES_DIR);
    await runNgNew(fixtureDirectory);

    process.chdir(path.join(FIXTURES_DIR, fixtureDirectory));
    await runNgAdd();
    await runNgGenerate(['app', 'another-app', '--interactive=false']);
    await runNgGenerate(['lib', 'another-lib', '--interactive=false']);
  });

  it('it should pass linting after creating a new workspace from scratch using @angular-eslint', async () => {
    // TSLint configs and dependencies should not be present
    expect(() =>
      require('../fixtures/v13-new-workspace/tslint.json'),
    ).toThrowErrorMatchingInlineSnapshot(
      `"Cannot find module '../fixtures/v13-new-workspace/tslint.json' from 'tests/v13-new-workspace.test.ts'"`,
    );
    expect(
      JSON.stringify(
        requireUncached('../fixtures/v13-new-workspace/package.json')
          .devDependencies,
        null,
        2,
      ),
    ).toMatchSnapshot();

    // Root project
    expect(
      requireUncached('../fixtures/v13-new-workspace/.eslintrc.json'),
    ).toMatchSnapshot();

    expect(
      requireUncached('../fixtures/v13-new-workspace/angular.json').projects[
        'v13-new-workspace'
      ].architect.lint,
    ).toMatchSnapshot();

    // Additional project ("another-app")
    expect(() =>
      require('../fixtures/v13-new-workspace/projects/another-app/tslint.json'),
    ).toThrowErrorMatchingInlineSnapshot(
      `"Cannot find module '../fixtures/v13-new-workspace/projects/another-app/tslint.json' from 'tests/v13-new-workspace.test.ts'"`,
    );
    expect(
      requireUncached(
        '../fixtures/v13-new-workspace/projects/another-app/.eslintrc.json',
      ),
    ).toMatchSnapshot();

    expect(
      requireUncached('../fixtures/v13-new-workspace/angular.json').projects[
        'another-app'
      ].architect.lint,
    ).toMatchSnapshot();

    // Additional library project ("another-lib")
    expect(() =>
      require('../fixtures/v13-new-workspace/projects/another-lib/tslint.json'),
    ).toThrowErrorMatchingInlineSnapshot(
      `"Cannot find module '../fixtures/v13-new-workspace/projects/another-lib/tslint.json' from 'tests/v13-new-workspace.test.ts'"`,
    );
    expect(
      requireUncached(
        '../fixtures/v13-new-workspace/projects/another-lib/.eslintrc.json',
      ),
    ).toMatchSnapshot();

    expect(
      requireUncached('../fixtures/v13-new-workspace/angular.json').projects[
        'another-lib'
      ].architect.lint,
    ).toMatchSnapshot();

    const lintOutput = await runLint(fixtureDirectory);
    expect(lintOutput).toMatchSnapshot();
  });
});
