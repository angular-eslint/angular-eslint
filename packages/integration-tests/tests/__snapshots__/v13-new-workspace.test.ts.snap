// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v13-new-workspace it should pass linting after creating a new workspace from scratch using @angular-eslint 2`] = `
Object {
  "@angular-devkit/build-angular": "^15.0.2",
  "@angular-eslint/builder": "9999.0.1-local-integration-tests",
  "@angular-eslint/eslint-plugin": "9999.0.1-local-integration-tests",
  "@angular-eslint/eslint-plugin-template": "9999.0.1-local-integration-tests",
  "@angular-eslint/schematics": "9999.0.1-local-integration-tests",
  "@angular-eslint/template-parser": "9999.0.1-local-integration-tests",
  "@angular/cli": "~15.0.2",
  "@angular/compiler-cli": "^15.0.0",
  "@types/jasmine": "~4.3.0",
  "@typescript-eslint/eslint-plugin": "5.44.0",
  "@typescript-eslint/parser": "5.44.0",
  "eslint": "^8.29.0",
  "jasmine-core": "~4.5.0",
  "karma": "~6.4.0",
  "karma-chrome-launcher": "~3.1.0",
  "karma-coverage": "~2.2.0",
  "karma-jasmine": "~5.1.0",
  "karma-jasmine-html-reporter": "~2.0.0",
  "ng-packagr": "^15.0.0",
  "typescript": "~4.8.2",
}
`;

exports[`v13-new-workspace it should pass linting after creating a new workspace from scratch using @angular-eslint 3`] = `
Object {
  "ignorePatterns": Array [
    "projects/**/*",
  ],
  "overrides": Array [
    Object {
      "extends": Array [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@angular-eslint/recommended",
        "plugin:@angular-eslint/template/process-inline-templates",
      ],
      "files": Array [
        "*.ts",
      ],
      "rules": Object {
        "@angular-eslint/component-selector": Array [
          "error",
          Object {
            "prefix": "app",
            "style": "kebab-case",
            "type": "element",
          },
        ],
        "@angular-eslint/directive-selector": Array [
          "error",
          Object {
            "prefix": "app",
            "style": "camelCase",
            "type": "attribute",
          },
        ],
      },
    },
    Object {
      "extends": Array [
        "plugin:@angular-eslint/template/recommended",
      ],
      "files": Array [
        "*.html",
      ],
      "rules": Object {},
    },
  ],
  "root": true,
}
`;

exports[`v13-new-workspace it should pass linting after creating a new workspace from scratch using @angular-eslint 4`] = `
Object {
  "builder": "@angular-eslint/builder:lint",
  "options": Object {
    "lintFilePatterns": Array [
      "src/**/*.ts",
      "src/**/*.html",
    ],
  },
}
`;

exports[`v13-new-workspace it should pass linting after creating a new workspace from scratch using @angular-eslint 6`] = `
Object {
  "extends": "../../.eslintrc.json",
  "ignorePatterns": Array [
    "!**/*",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        "*.ts",
      ],
      "rules": Object {
        "@angular-eslint/component-selector": Array [
          "error",
          Object {
            "prefix": "app",
            "style": "kebab-case",
            "type": "element",
          },
        ],
        "@angular-eslint/directive-selector": Array [
          "error",
          Object {
            "prefix": "app",
            "style": "camelCase",
            "type": "attribute",
          },
        ],
      },
    },
    Object {
      "files": Array [
        "*.html",
      ],
      "rules": Object {},
    },
  ],
}
`;

exports[`v13-new-workspace it should pass linting after creating a new workspace from scratch using @angular-eslint 7`] = `
Object {
  "builder": "@angular-eslint/builder:lint",
  "options": Object {
    "lintFilePatterns": Array [
      "projects/another-app/**/*.ts",
      "projects/another-app/**/*.html",
    ],
  },
}
`;

exports[`v13-new-workspace it should pass linting after creating a new workspace from scratch using @angular-eslint 9`] = `
Object {
  "extends": "../../.eslintrc.json",
  "ignorePatterns": Array [
    "!**/*",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        "*.ts",
      ],
      "rules": Object {
        "@angular-eslint/component-selector": Array [
          "error",
          Object {
            "prefix": "lib",
            "style": "kebab-case",
            "type": "element",
          },
        ],
        "@angular-eslint/directive-selector": Array [
          "error",
          Object {
            "prefix": "lib",
            "style": "camelCase",
            "type": "attribute",
          },
        ],
      },
    },
    Object {
      "files": Array [
        "*.html",
      ],
      "rules": Object {},
    },
  ],
}
`;

exports[`v13-new-workspace it should pass linting after creating a new workspace from scratch using @angular-eslint 10`] = `
Object {
  "builder": "@angular-eslint/builder:lint",
  "options": Object {
    "lintFilePatterns": Array [
      "projects/another-lib/**/*.ts",
      "projects/another-lib/**/*.html",
    ],
  },
}
`;

exports[`v13-new-workspace it should pass linting after creating a new workspace from scratch using @angular-eslint 11`] = `
"
Linting \\"v13-new-workspace\\"...

All files pass linting.


Linting \\"another-app\\"...

All files pass linting.


Linting \\"another-lib\\"...

__ROOT__/v13-new-workspace/projects/another-lib/src/lib/another-lib.service.ts
  8:17  error  Unexpected empty constructor  @typescript-eslint/no-empty-function

âœ– 1 problem (1 error, 0 warnings)
"
`;
